File: generate_icon.py
#!/usr/bin/env python3
"""
Generate VoiceIt app icon with waveform symbol
"""

from PIL import Image, ImageDraw, ImageFont
import os

def create_gradient(width, height, color1, color2):
    """Create a vertical gradient"""
    base = Image.new('RGB', (width, height), color1)
    top = Image.new('RGB', (width, height), color2)
    mask = Image.new('L', (width, height))
    mask_data = []
    for y in range(height):
        mask_data.extend([int(255 * (y / height))] * width)
    mask.putdata(mask_data)
    base.paste(top, (0, 0), mask)
    return base

def draw_waveform(draw, center_x, center_y, size, color):
    """Draw a simple waveform icon"""
    bar_width = size // 10
    spacing = size // 12
    
    # 5 vertical bars of varying heights
    bars = [
        (center_x - 2 * (bar_width + spacing), center_y, 0.5),  # Left
        (center_x - (bar_width + spacing), center_y, 0.7),
        (center_x, center_y, 1.0),  # Center (tallest)
        (center_x + (bar_width + spacing), center_y, 0.7),
        (center_x + 2 * (bar_width + spacing), center_y, 0.5),  # Right
    ]
    
    for x, y, height_ratio in bars:
        bar_height = size * height_ratio
        top = y - bar_height // 2
        bottom = y + bar_height // 2
        draw.rounded_rectangle(
            [x - bar_width // 2, top, x + bar_width // 2, bottom],
            radius=bar_width // 2,
            fill=color
        )

def create_app_icon(output_path, size=1024):
    """Create the VoiceIt app icon"""
    # Purple gradient colors (matching app theme)
    color1 = (124, 58, 237)  # #7C3AED (voiceitPurple)
    color2 = (167, 85, 255)  # Lighter purple
    
    # Create gradient background (RGB, fully opaque - iOS adds rounded corners automatically)
    img = create_gradient(size, size, color1, color2)
    draw = ImageDraw.Draw(img)
    
    # Draw waveform symbol in white
    waveform_size = size // 2
    draw_waveform(draw, size // 2, size // 2, waveform_size, (255, 255, 255))
    
    # Save as PNG without alpha channel (fully opaque)
    img.save(output_path, 'PNG')
    print(f"✅ Created app icon: {output_path}")

if __name__ == '__main__':
    # Generate the 1024x1024 icon required by iOS
    icon_path = 'VoiceIt/Assets.xcassets/AppIcon.appiconset/icon-1024.png'
    
    # Check if PIL is installed
    try:
        create_app_icon(icon_path, 1024)
        print(f"\n🎉 App icon generated successfully!")
        print(f"📱 The icon will appear when you build the app.")
    except Exception as e:
        print(f"❌ Error: {e}")
        print(f"\n💡 To fix this, install Pillow:")
        print(f"   pip3 install Pillow")

File: generate_icon.swift
#!/usr/bin/env swift

import Foundation
import AppKit

func createGradient(size: CGSize, color1: NSColor, color2: NSColor) -> NSImage {
    let image = NSImage(size: size)
    image.lockFocus()
    
    let gradient = NSGradient(starting: color1, ending: color2)
    gradient?.draw(in: NSRect(origin: .zero, size: size), angle: 90)
    
    image.unlockFocus()
    return image
}

func drawWaveform(in context: CGContext, center: CGPoint, size: CGFloat, color: NSColor) {
    let barWidth = size / 10
    let spacing = size / 12
    
    // 5 vertical bars of varying heights
    let bars: [(CGFloat, CGFloat)] = [
        (center.x - 2 * (barWidth + spacing), 0.5),  // Left
        (center.x - (barWidth + spacing), 0.7),
        (center.x, 1.0),  // Center (tallest)
        (center.x + (barWidth + spacing), 0.7),
        (center.x + 2 * (barWidth + spacing), 0.5),  // Right
    ]
    
    context.setFillColor(color.cgColor)
    
    for (x, heightRatio) in bars {
        let barHeight = size * heightRatio
        let top = center.y - barHeight / 2
        let rect = CGRect(x: x - barWidth / 2, y: top, width: barWidth, height: barHeight)
        let path = NSBezierPath(roundedRect: rect, xRadius: barWidth / 2, yRadius: barWidth / 2)
        context.addPath(path.cgPath)
        context.fillPath()
    }
}

func createAppIcon(size: CGFloat) -> NSImage? {
    // Purple gradient colors
    let color1 = NSColor(red: 124/255, green: 58/255, blue: 237/255, alpha: 1.0)
    let color2 = NSColor(red: 167/255, green: 85/255, blue: 255/255, alpha: 1.0)
    
    let imageSize = NSSize(width: size, height: size)
    let image = NSImage(size: imageSize)
    
    image.lockFocus()
    
    // Draw gradient background
    let gradient = NSGradient(starting: color1, ending: color2)
    let cornerRadius = size / 5
    let path = NSBezierPath(roundedRect: NSRect(origin: .zero, size: imageSize), 
                           xRadius: cornerRadius, 
                           yRadius: cornerRadius)
    path.addClip()
    gradient?.draw(in: NSRect(origin: .zero, size: imageSize), angle: 90)
    
    // Draw waveform
    if let context = NSGraphicsContext.current?.cgContext {
        let center = CGPoint(x: size / 2, y: size / 2)
        let waveformSize = size / 2
        drawWaveform(in: context, center: center, size: waveformSize, color: .white)
    }
    
    image.unlockFocus()
    return image
}

// Generate the icon
let size: CGFloat = 1024
if let icon = createAppIcon(size: size) {
    if let tiffData = icon.tiffRepresentation,
       let bitmap = NSBitmapImageRep(data: tiffData),
       let pngData = bitmap.representation(using: .png, properties: [:]) {
        
        let outputPath = "VoiceIt/Assets.xcassets/AppIcon.appiconset/icon-1024.png"
        let url = URL(fileURLWithPath: outputPath)
        
        do {
            try pngData.write(to: url)
            print("✅ Created app icon: \(outputPath)")
            print("🎉 App icon generated successfully!")
            print("📱 The icon will appear when you build the app.")
        } catch {
            print("❌ Error writing file: \(error)")
        }
    }
} else {
    print("❌ Failed to create icon")
}

